libobs/obs-cocoa.m

```diff
@@ -1804,3 +1804,17 @@ bool obs_hotkeys_platform_is_pressed(obs_hotkeys_platform_t *plat,
 
 	return false;
 }
+
+void *obs_graphics_thread_mac(void *param)
+{
+	@autoreleasepool {
+		return obs_graphics_thread(param);
+	}
+}
+
+bool obs_graphics_thread_loop_mac(struct obs_graphics_context *context)
+{
+	@autoreleasepool {
+		return obs_graphics_thread_loop(context);
+	}
+}
```

Comment:
I don't know how OBS calls these, but the pool needs to be part of the the loop, eg:

```c++
while (true) {
    @autoreleasepool {
        do_stuff_per_iteration();
    }
}
```
Otherwise the pool will only be drained once the thread exits.

Reply:
Never mind, I see you're calling `obs_graphics_thread_loop_mac` in a loop below.